grpc/src/main/java/io/stargate/grpc/payload/cql/ValuesHandler.java

```diff
@@ -0,0 +1,169 @@
+package io.stargate.grpc.payload.cql;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.grpc.Status;
+import io.grpc.StatusException;
+import io.stargate.db.BoundStatement;
+import io.stargate.db.Result.Prepared;
+import io.stargate.db.Result.Rows;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.Column.ColumnType;
+import io.stargate.db.schema.UserDefinedType;
+import io.stargate.grpc.codec.cql.ValueCodec;
+import io.stargate.grpc.codec.cql.ValueCodecs;
+import io.stargate.grpc.payload.PayloadHandler;
+import io.stargate.proto.QueryOuterClass.*;
+import io.stargate.proto.QueryOuterClass.Payload.Type;
+import io.stargate.proto.QueryOuterClass.TypeSpec.Builder;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ValuesHandler implements PayloadHandler {
+  @Override
+  public BoundStatement bindValues(Prepared prepared, Payload payload)
+      throws InvalidProtocolBufferException, StatusException {
+    final Values values = payload.getValue().unpack(Values.class);
+    final List<Column> columns = prepared.metadata.columns;
+    final int columnCount = columns.size();
+    if (values == null && columnCount > 0 || columnCount != values.getValuesCount()) {
+      throw Status.FAILED_PRECONDITION
+          .withDescription(
+              String.format(
+                  "Invalid number of bind parameters. Expected %d, but received %d",
+                  columnCount, values == null ? 0 : values.getValuesCount()))
+          .asException();
+    }
+    final List<ByteBuffer> boundValues = new ArrayList<>(columnCount);
+    List<String> boundValueNames = null;
+    if (values.getValueNamesCount() != 0) {
+      final int namesCount = values.getValueNamesCount();
+      boundValueNames = new ArrayList<>(namesCount);
+      for (int i = 0; i < namesCount; ++i) {
+        String name = values.getValueNames(i);
+        Column column =
+            columns.stream()
+                .filter(c -> c.name() == name)
```

Comment:
Seems you are comparing String by reference equality here?

Reply:
Thanks. Good catch.